### PR TITLE
Fix: autosuggest breaks search blocks

### PR DESCRIPTION
**Pull request by dinhtungdu**
### Description of the Change
Autosuggest changes the frontend markup of the search field to add the suggestion list, while it works for the normal search field, it's breaking the search block styling due to CSS conflict between autosuggest and search block. Because the markup of the search block is predictable, this PR adds a separated behavior for search field in the search blocks that reuses the block markup to add the suggestion list and required class.

### Alternate Designs
n/a

### Benefits
Make autosuggest work with search block, which is a preparation for full site editing.

### Possible Drawbacks
n/a

### Verification Process
1. Enable and configured Autosuggest features.
2. Create a new page and add a search block.
3. View the page on the front end, type a keyword, see the suggestions.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues
#2211

### Changelog Entry
Fixed the styling issue of Autosuggest and search block.

### CODE REVIEW

**Comment by dinhtungdu** [comment]
Some screenshots:

### The breaking style:
Screenshot at May 27 11-33-50
---
### The style after applying this PR:
image

**Comment by Rahmon** [comment]
Hi @dinhtungdu 

When the result has more than one item, the list appears cut and I can see just the first item. 

https://user-images.githubusercontent.com/6010232/120329957-ba4d3c80-c2c2-11eb-933f-6b3eb7c317c4.mp4

**Comment by dinhtungdu** [comment]
@Rahmon Look like you used 2020 for your test. That issue causes by 2020 styling which set `overflow: hidden;` to the `<main>` element:
image

It really depends on the theme styling to fix that issue. The issue doesn't occur on Storefront for example: 
image

**Comment by Rahmon** [comment]
@dinhtungdu got you.

Looks good to me. But I would like to have second thought here so I'm adding @JakePT as a reviewer too since he can contribute with your frontend knowledge.

File: assets/js/autosuggest.js [hunk 1]
```diff
@@ -432,6 +440,24 @@ function init() {
...
 
+	/**
+	 * For search blocks, because we know the output mark up, we reuse it
+	 * for autosuggest.
+	 */
+	epBlockInputs.forEach((input) => {
+		// Disable autocomplete
+		input.setAttribute('autocomplete', 'off');
+
+		input.closest('form').classList.add('ep-autosuggest-container');
```
Comment (JakePT) [inline assets/js/autosuggest.js:451]: You can get the form element that an input belongs to with the `form` property:
```
input.form.classList.add('ep-autosuggest-container');
```

**Review by JakePT — COMMENTED** [review]
@dinhtungdu Just added a small suggestion.

**Comment by dinhtungdu** [comment]
@JakePT Thanks so much for the tip! @Rahmon I force-pushed the branch to clean the history and the diff. This PR is ready to review again.

**Review by Rahmon — APPROVED** [review]

**Review by JakePT — APPROVED** [review]

**Pull request merged**